Namespace for sbus scale rules

Larocceau opened the pull request:
This PR closes #1188 

The changes in this PR are as follows:

* Adds the field namespace to the ServiceBusScaleRule, which is used to associate the scale rule to the right sbus


I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [ ] **Tested my code** end-to-end against a live Azure subscription.
* [x] **Updated the documentation** in the docs folder for the affected changes.
* [x] **Written unit tests** against the modified code that I have made.
* [x] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [x] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

* Instead of testing on a existing deployment, I compared a ARM template of a successful implementation of this feature using a custom scale rule, with one using this feature. There was no significant difference in the two templates.

Below is a minimal example configuration that includes the new features, which can be used to deploy to Azure:

```fsharp
containerApp {
      ....
      add_servicebus_scale_rule "sb-keda-scale" {
          QueueName = "wishrequests"
          MessageCount = 5
          SecretRef = "servicebusconnectionkey"
          Namespace = "servicebus"
      }
}
```

Larocceau committed:
add namespace to service bus scale rule

Larocceau committed:
update release notes

Larocceau committed:
fix container app test

mattgallagher92 commented on src/Tests/ContainerApps.fs:
Should one of the tests check that the value is set correctly?

mattgallagher92 commented on RELEASE_NOTES.md:
Do the docs need updating too?

mattgallagher92 reviewed (commented):
Is this field optional? If so, this change will break any existing clients, so we should probably make the new field optional too. If it's mandatory then there can't be any existing clients so we don't have to worry about that, in which case this change is fine (but I've added some comments about some things to improve).

Larocceau commented:
Hi @ninjarobot! Sorry for the delay; things been a bit messy for us, and I've left CIT (as has most other dev staff). I did not get around this, but would be happy to wrap this up later.

Copilot committed:
Merge branch 'master' of https://github.com/CompositionalIT/farmer into namespace-for-sbus-scale-rules

# Conflicts:
#	RELEASE_NOTES.md

Co-authored-by: ninjarobot <1520226+ninjarobot@users.noreply.github.com>

Copilot committed:
Add namespace test assertion and docs update for ServiceBusScaleRule

Agent-Logs-Url: https://github.com/CompositionalIT/farmer/sessions/7d45065b-a4c6-4a71-8b8d-d805f0d13ad7

Co-authored-by: ninjarobot <1520226+ninjarobot@users.noreply.github.com>
